crates/bevy_pbr/src/light/mod.rs

```diff
@@ -789,6 +803,18 @@ pub fn check_dir_light_mesh_visibility(
                 .for_each(shrink_entities);
         }
     }
+
+    // TODO: use resource to avoid unnecessary memory alloc
+    let mut defer_queue = std::mem::take(visible_entities_queue.deref_mut());
```

Comment:
```suggestion
    // Defer marking view visibility so this system can run in parallel with check_point_light_mesh_visibility
    // TODO: use resource to avoid unnecessary memory alloc
    let mut defer_queue = std::mem::take(visible_entities_queue.deref_mut());
```